client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -14,23 +14,63 @@ import { connect } from 'react-redux';
 import Card from 'components/card';
 import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
 import FormFieldset from 'components/forms/form-fieldset';
+import CompactFormToggle from 'components/forms/form-toggle/compact';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
 import isJetpackModuleUnavailableInDevelopmentMode from 'state/selectors/is-jetpack-module-unavailable-in-development-mode';
 import isJetpackSiteInDevelopmentMode from 'state/selectors/is-jetpack-site-in-development-mode';
+import { activateModule, deactivateModule } from 'state/jetpack/modules/actions';
+import isJetpackModuleActive from 'state/selectors/is-jetpack-module-active';
+import isActivatingJetpackModule from 'state/selectors/is-activating-jetpack-module';
+import isDeactivatingJetpackModule from 'state/selectors/is-deactivating-jetpack-module';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { getSiteSlug } from 'state/sites/selectors';
 import SupportInfo from 'components/support-info';
 
 class SpeedUpSiteSettings extends Component {
+	static defaultProps = {
+		togglingSiteAccelerator: false,
+	};
+
 	static propTypes = {
 		fields: PropTypes.object,
 		isRequestingSettings: PropTypes.bool,
 		isSavingSettings: PropTypes.bool,
 		jetpackVersionSupportsLazyImages: PropTypes.bool,
+		siteAcceleratorSupported: PropTypes.bool,
+		activateModule: PropTypes.func,
+		deactivateModule: PropTypes.func,
 
 		// Connected props
 		photonModuleUnavailable: PropTypes.bool,
 		selectedSiteId: PropTypes.number,
 		siteSlug: PropTypes.string,
+		photonModuleActive: PropTypes.bool,
+		assetCdnModuleActive: PropTypes.bool,
+		togglingSiteAccelerator: PropTypes.bool,
+	};
+
+	handleCdnChange = () => {
+		const { photonModuleActive, assetCdnModuleActive, selectedSiteId } = this.props;
+
+		// Check if any of the CDN options are on.
+		const cdnStatus = photonModuleActive || assetCdnModuleActive;
+
+		// If one of them is on, we turn everything off.
+		if ( true === cdnStatus ) {
```

Comment:
All of this logic seems a bit repetitive - do you think we can figure a helper function to avoid this?

Reply:
That's an issue I've had with the interface in Jetpack as well, but I can't really figure out how to make it simpler. I also don't imagine we'll have many, if any, other modules that behave that way?

Maybe it would be worth revisiting once we have more toggles that activate / deactivate multiple modules at once?

Reply:
I keep thinking that there are too many ways this could go wrong. We're doing several requests for this, and what do we do if one of them fails? We might even not have the latest module activation state in Calypso at the time of toggling. Seems like this is fragile and should be handled on the Jetpack side. What do you think?

Also see https://github.com/Automattic/wp-calypso/pull/27944#discussion_r230758374